### ISSUE TITLE
Modularize Code

### ISSUE THREAD

**Issue by lchski** [lchski/flip-coin#1]
Code is currently very procedural and hurts my eyes to look at. Need to modularize it a bit more and prettify the code better (cleaning up the code is a better term).

**Comment by johnafish** [comment]
Yeah, our variables are all over the place right now, and that's probably the smallest problem. We need to change the way that the ask for name system works; as currently it hurts my brain to think about. We'll just have it return null if N. I'll hop on skype and talk to you for a bit if you aren't busy.

**Comment by jalotra** [comment]
Haha, you did skype call over this?
